libs/core/src/lib/carousel/carousel-item/carousel-item.component.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    HostBinding,
+    Input,
+    ViewEncapsulation
+} from '@angular/core';
+
+let carouselItemUniqueId = 0;
+
+@Component({
+    selector: 'fd-carousel-item',
+    templateUrl: './carousel-item.component.html',
+    styleUrls: ['./carousel-item.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class CarouselItemComponent {
```

Comment:
Base component extension won't help here?

Reply:
Base component is in platform. carousel is in core. so will not be able to use base component.